### PR TITLE
Store states as files in DefaultStore

### PR DESCRIPTION
**Pull request by longfin**
This PR separates states from `index.ldb` to help maintain with large states. 

also, I skipped changelog because 0.8.0 introducing `DefaultStore` wasn't released yet. 😕

### CODE REVIEW

File: Libplanet/Store/DefaultStore.cs [hunk 1]
```diff
@@ -551,7 +571,6 @@ HashDigest<SHA256> blockHash
             HashDigest<SHA256> blockHash,
             IImmutableDictionary<Address, IValue> states)
         {
-            _statesCache.AddOrUpdate(blockHash, states);
```
Comment (moreal) [inline Libplanet/Store/DefaultStore.cs]: It seems caching after calling `GetBlockStates()` at least once, is it intended?
If then, I wonder why `SetBlockStates()` doesn't set cache. 🤔

Comment (moreal) [reply Libplanet/Store/DefaultStore.cs]: Oh sorry, it was moved to line 603. It was my mistake.

File: Libplanet/Store/DefaultStore.cs [hunk 2]
```diff
@@ -524,20 +528,36 @@ HashDigest<SHA256> blockHash
...
+                    }
+                }
+                else
+                {
+                    file.CopyTo(buffer);
+                }
 
-                var deserialized = (Bencodex.Types.Dictionary)_codec.Decode(stream);
-                ImmutableDictionary<Address, IValue> states = deserialized.ToImmutableDictionary(
+                IValue value = new Codec().Decode(buffer.ToArray());
```
Comment (dahlia) [inline Libplanet/Store/DefaultStore.cs]: `Bencodex.Codec` has an overload `Decode(Stream)` too.  You could reuse the `buffer` object by `buffer.Seek(0)` without allocating a new array.

File: Libplanet/Store/DefaultStore.cs [hunk 3]
```diff
@@ -560,16 +579,28 @@ HashDigest<SHA256> blockHash
                     )
                 )
             );
-            using (var stream = new MemoryStream())
+
+            UPath path = StatePath(blockHash);
+            UPath dirPath = path.GetDirectory();
+            CreateDirectoryRecursively(_states, dirPath);
+
+            using (var buffer = new MemoryStream(new Codec().Encode(serialized)))
```
Comment (dahlia) [inline Libplanet/Store/DefaultStore.cs]: As `Bencodex.Codec` has an overload `Encode(IValue, Stream)` as well, you don't have a `buffer` but could directly write to `deflate` or `file` instead, e.g., `new Codec().Encode(serialized, deflate)`.

**Pull request merged**